packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -652,6 +685,83 @@ class Auth {
         return token
     }
 
+    /**
+     * TODO: TAOB Eventually this will be a wrapper method for commerce-sdk-isomorphic helper: authorizeTrustedAgent.
+     *
+     */
+    async authorizeTrustedAgent(credentials: {loginId?: string}) {
+        const slasClient = this.client
+        const codeVerifier = helpers.createCodeVerifier()
+        const codeChallenge = await helpers.generateCodeChallenge(codeVerifier)
+        const organizationId = slasClient.clientConfig.parameters.organizationId
+        const clientId = slasClient.clientConfig.parameters.clientId
+        const siteId = slasClient.clientConfig.parameters.siteId
+        const loginId = credentials.loginId || 'guest'
+        const idpOrigin = loginId === 'guest' ? 'slas' : 'ecom'
+
+        const url = [
+            `${slasClient.clientConfig.proxy}/shopper/auth/v1/organizations/${organizationId}/oauth2/trusted-agent/authorize`,
+                `?client_id=${clientId}`,
+                `&channel_id=${siteId}`,
+                `&code_challenge=${codeChallenge}`,
+                `&login_id=${loginId}`,
+                `&redirect_uri=${this.redirectURI}`,
+                `&idp_origin=${idpOrigin}`,
+                `&response_type=code`
+            ].join('')
+
+        return {url, codeVerifier}
```

Comment:
@joeluong-sfcc this is where we're thinking the "URL builder" aspect should possibly live in `commerce-sdk-isomorphic`, I'm assuming this is an anti-pattern to not derive API paths from that library